Tasks/AzureKeyVault/operations/KeyVault.ts

```diff
@@ -0,0 +1,133 @@
+                    console.log(tl.loc("NoSecretsFound", this.taskParameters.keyVaultName));
+                    return resolve();
+                }
+
+                console.log(tl.loc("NumberOfSecretsFound", this.taskParameters.keyVaultName, listOfSecrets.length));
+                listOfSecrets = this.filterDisabledAndExpiredSecrets(listOfSecrets);
+                console.log(tl.loc("NumberOfEnabledSecretsFound", this.taskParameters.keyVaultName, listOfSecrets.length));
+
+                var getSecretValuePromises: Promise<any>[] = [];
+                listOfSecrets.forEach((secret: armKeyVault.AzureKeyVaultSecret, index: number) => {
+                    getSecretValuePromises.push(this.downloadSecretValue(secret.name));
+                });
+
+                Promise.all(getSecretValuePromises).then(() =>{
+                    return resolve();
+                }).catch((error) => {
+                    return reject(error);
+                });
+            });
+        });
+    }
+
+    private downloadSelectedSecrets(selectedSecrets: string[]): Promise<void> {
+        tl.debug(util.format("Downloading selected secrets from vault: %s", this.taskParameters.keyVaultName));
+
+        return new Promise<void>((resolve, reject) => {
+            var getSecretValuePromises: Promise<any>[] = [];
+            selectedSecrets.forEach((secretName: string, index: number) => {
+                getSecretValuePromises.push(this.downloadSecretValue(secretName));
+            });
+
+            Promise.all(getSecretValuePromises).then(() =>{
+                return resolve();
+            }).catch((error) => {
+                return reject(error);
+            });
+        });
+    }
+
+    private filterDisabledAndExpiredSecrets(listOfSecrets: armKeyVault.AzureKeyVaultSecret[]): armKeyVault.AzureKeyVaultSecret[] {
+        var result: armKeyVault.AzureKeyVaultSecret[] = [];
+        var now: Date = new Date();
+
+        listOfSecrets.forEach((value: armKeyVault.AzureKeyVaultSecret, index: number) => {
+            if (value.enabled && (!value.expires || value.expires > now)) {
+                result.push(value);
+            }
+        });
+        
+        return result;
+    }
+
+    private downloadSecretValue(secretName: string): Promise<any> {
+        tl.debug(util.format("Downloading secret value for: %s", secretName));
+
+        return new Promise<void>((resolve, reject) => {
+            // if (tl.getVariable(secretName) !== undefined) {
+            //     return reject(tl.loc("ConflictingVariableFound", secretName));
+            // }
+
+            this.keyVaultClient.getSecretValue(secretName, (error, secretValue, request, response) => {
+                if (error) {
+                    return reject(tl.loc("GetSecretValueFailed", secretName, this.getError(error)));
+                }
+                
+                console.log("##vso[task.setvariable variable=" + secretName + ";issecret=true;]" + secretValue);
```

Comment:
If secretName and secretValue can support special characters, please test if you are able to set names with those characters in environment variable.